api-guidelines/v4.0/API-DESIGN-GUIDELINES/GET-REQUESTS.md

```diff
@@ -0,0 +1,307 @@
+# GET Requests
+
+An HTTP `GET` returns existing Resource item(s). `GET` _must_ be an idempotent
+operation. `GET` is a required verb.
+
+## Request URLs
+
+A `GET` request can specify either a collection of items or a specific item:
+
+* `GET /ed-fi/students` with or without query string parameters retrieves a list
+  ("GET All" queries).
+* `GET /ed-fi/students/[identifier]` retrieves a specific student ("GET by ID").
+
+If the natural key is provided as a query string, then the result will be a
+collection containing only one item.
+
+An Ed-Fi API _should_ support searching capabilities with the possible use of
+filters, paging, and ordering. These are discussed below.  Information about all
+supported capabilities, their configurations, and their default values should be
+available to client applications via the [Discovery API](./DISCOVERY-API.md).
+
+### Search
+
+An Ed-Fi API _should_ support querying capabilities when searching a collection
+of Resources. Query operators are applied to the query string using the
```

Comment:
_should_ conflicts with the section in natural/foreign keys where GETs by natural key are required. Would clarify if the intent is just that fields not used as a natural key are optional.